### ISSUE TITLE
WebGPU `requestAdapter` failure isn't handled

### ISSUE THREAD

**Issue by kungfooman** [playcanvas/engine#5437]
![image](https://github.com/playcanvas/engine/assets/5236548/4c6eef7c-ce9d-4a4e-9cf1-9c461ef502a8)

This messes up the examples for me too, because they default to WebGPU:

![image](https://github.com/playcanvas/engine/assets/5236548/9c9b5461-3fe5-40e7-b6a3-fd6e787d6039)

I'm on Linux/Chrome, which still doesn't support WebGPU :hourglass: :eyes: 

But it's exporting `navigator.gpu` anyway:

![image](https://github.com/playcanvas/engine/assets/5236548/5a693999-c81e-4494-a324-c3103fe733c5)

**Comment by kungfooman** [comment]
@ellthompson Can you set default renderer to WebGL2?

You can do so via `value: DEVICETYPE_WEBGL2` in `examples/src/app` `Example` class:

```js
                  <SelectInput
                    id='deviceTypeSelectInput'
                    options={[
                        { t: deviceTypeNames[DEVICETYPE_WEBGL1], v: DEVICETYPE_WEBGL1 },
                        { t: deviceTypeNames[DEVICETYPE_WEBGL2], v: DEVICETYPE_WEBGL2 },
                        { t: deviceTypeNames[DEVICETYPE_WEBGPU], v: DEVICETYPE_WEBGPU }
                    ]}
                    value={DEVICETYPE_WEBGL2}
                    onSelect={this.onSetPreferredGraphicsDevice}
                    prefix='Active Device: '
                    // @ts-ignore this is setting a legacy ref
                    ref={this.deviceTypeSelectInputRef}
                />
```

**Comment by mvaligursky** [comment]
Ideally we'd want to handle requestAdapter returning null, and fallback to WebGL2 device, but that's not straightforward.

**Comment by mvaligursky** [comment]
This should be fixed by #5721
